### PR TITLE
{vis}[GCCcore/9.3.0] TTK v0.9.8.9, zfp v0.5.5, PyQt5 v5.13.2 w/ Python 3.8.2

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
@SimonPinches Please add the missing checksums.

**Comment by SimonPinches** [comment]
@verdurin I've tried to add the missing checksums with `eb PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb --inject-checksums -f` but despite the promising listing, the actual EB file is unchanged:
```
$ eb PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb --inject-checksums -f
== temporary log file in case of crash /tmp/eb-vwvnw6/easybuild-y0u_gW.log
== found valid index for /work/imas/opt/EasyBuild/software/EasyBuild/4.2.1/easybuild/easyconfigs, so using it...
== injecting sha256 checksums in /tmp/pinches/easybuild-easyconfigs/easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb
== fetching sources & patches for PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb...
== backup of easyconfig file saved to /tmp/pinches/easybuild-easyconfigs/easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb.bak_20200616130051_20252...
== injecting sha256 checksums for sources & patches in PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb...
== * sip-4.19.22.tar.gz: e1b768824ec1a2ee38dd536b6b6b3d06de27b00a2f5f55470d1b512306e3be45
== * PyQt5-5.13.2.tar.gz: adc17c077bf233987b8e43ada87d1e0deca9bd71a13e5fd5fc377482ed69c827
== * PyQtWebEngine-5.13.2.tar.gz: 4264911b5847c75721d8c9c30af92e58a216bd25ceef37f7abf921005c1d45a9
== Temporary log file(s) /tmp/eb-vwvnw6/easybuild-y0u_gW.log* have been removed.
== Temporary directory /tmp/eb-vwvnw6 has been removed.
```

File: easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    "--no-stubs",
+])
+
+local_setup_env = "export PATH=%(installdir)s/bin:$PATH && "
+local_setup_env += "export PYTHONPATH=%(installdir)s/lib/python%(pyshortver)s/site-packages:$PYTHONPATH && "
+local_sipver = '4.19.22'
+components = [
+    ('SIP', local_sipver, {
+        'source_urls': ['https://www.riverbankcomputing.com/static/Downloads/sip/%s' % local_sipver],
+        'sources': [SOURCELOWER_TAR_GZ],
```
Comment (lexming) [inline easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb:67]:
```suggestion
        'sources': [SOURCELOWER_TAR_GZ],
        'checksums': ['e1b768824ec1a2ee38dd536b6b6b3d06de27b00a2f5f55470d1b512306e3be45'],
```

File: easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        'start_dir': 'sip-%s' % local_sipver,
+        'configopts': local_sip_configopts,
+        'options': {'modulename': 'PyQt5.sip'},
+    }),
+    (name, version, {
+        'source_urls': ['https://www.riverbankcomputing.com/static/Downloads/%(name)s/%(version)s/'],
+        'sources': ['%(name)s-%(version)s.tar.gz'],
```
Comment (lexming) [inline easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb:74]:
```suggestion
        'sources': ['%(name)s-%(version)s.tar.gz'],
        'checksums': ['adc17c077bf233987b8e43ada87d1e0deca9bd71a13e5fd5fc377482ed69c827'],
```

File: easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        'start_dir': '%(name)s-%(version)s',
+        'preconfigopts': local_setup_env,
+        'configopts': local_pyqt5_configopts,
+        'preinstallopts': local_setup_env,
+        'options': {'modulename': 'PyQt5.QtCore'},
+    }),
+    ('PyQtWebEngine', version, {
+        'source_urls': ['https://www.riverbankcomputing.com/static/Downloads/%(name)s/%(version)s/'],
+        'sources': ['%(name)s-%(version)s.tar.gz'],
```
Comment (lexming) [inline easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb:83]:
```suggestion
        'sources': ['%(name)s-%(version)s.tar.gz'],
        'checksums': ['4264911b5847c75721d8c9c30af92e58a216bd25ceef37f7abf921005c1d45a9'],
```

File: easybuild/easyconfigs/t/TTK/TTK-0.9.8.9-foss-2018a-Python-3.6.4.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+toolchainopts = {'pic': True, 'usempi': True}
+
+source_urls = ['https://github.com/topology-tool-kit/%(namelower)s/archive']
+sources = ["%(version)s.tar.gz"]
+checksums = ['d0cd86bbea9e761b0a9acb3ce871e5faaec17ee86c608fe6f472aa21399c31f5']
+
+dependencies = [
+    ('Boost', '1.66.0'),
+    ('Python', '3.6.4'),
+    ('VTK', '8.2.0', versionsuffix),
```
Comment (lexming) [inline easybuild/easyconfigs/t/TTK/TTK-0.9.8.9-foss-2018a-Python-3.6.4.eb:20]: You can make any modifications in your local installation and add an additional version of VTK to `2018a`. However, we restrict dependencies in the central repository of easyconfigs to one per toolchain to keep a manageable maintenance cost. In the case of `foss/2018a` the version of VTK available is `8.1.0`. Therefore, this easyconfig of TTK should use that dependency to be accepted.

Comment (jaswantp) [reply easybuild/easyconfigs/t/TTK/TTK-0.9.8.9-foss-2018a-Python-3.6.4.eb:20]: This breaks python bindings of TTK. In that, the bindings are successfully generated only with VTK >= 8.2.

Comment (lexming) [reply easybuild/easyconfigs/t/TTK/TTK-0.9.8.9-foss-2018a-Python-3.6.4.eb:20]: In such a case, if you want to contribute this easyconfig for `TTK`, please consider moving it to either `foss/2019a`, `foss/2019b` or `foss/2020a`. All those toolchains provide VTK v8.2.

File: easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb (hunk 1 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb:67]: Thanks @lexming - I've now done this.

File: easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb (hunk 3 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/p/PyQt5/PyQt5-5.13.2-GCCcore-9.3.0-Python-3.8.2.eb:85]: Thanks @lexming - now done.

File: easybuild/easyconfigs/t/TTK/TTK-0.9.8.9-foss-2018a-Python-3.6.4.eb (hunk 4 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/t/TTK/TTK-0.9.8.9-foss-2018a-Python-3.6.4.eb]: @lexming, I've now removed the offending EB files.

**Comment by akesandgren** [comment]
@SimonPinches Can you update this to use the already merged PyQt5/5.15.1 we have for GCCcore/9.3.0?

**Comment by akesandgren** [comment]
@SimonPinches Ping regarding using the already merged PyQt5/5.15.1 for GCCcore/9.3.0 !!

**Comment by akesandgren** [comment]
On the other hand this toolchain has been deprecated, you'll need to update it to a newer one.

**Comment by akesandgren** [comment]
Closing this PR, since it (only) uses deprecated toolchain versions,
see https://docs.easybuild.io/policies/toolchains .

**Pull request closed without merging**